api/v1beta1/module_types.go

```diff
@@ -60,7 +60,8 @@ type Build struct {
 	// BuildArgs is an array of build variables that are provided to the image building backend.
 	BuildArgs []BuildArg `json:"buildArgs"`
 
-	Dockerfile string `json:"dockerfile"`
+	// ConfigMap that holds Dockerfile contents
+	DockerfileRef *v1.LocalObjectReference `json:"dockerfileRef"`
```

Comment:
I think this will cause an issue:
Change to the config map won't trigger a reconcile.

Reply:
@yevgeny-shnaidman can we leave `Dockerfile` as an option? It has the benefit of not breaking the API, which requires a version bump and a conversion webhook...

Reply:
we don't want to be triggered by ConfigMap. If the image is already build, then we won't run the Build flow again, and if the build has failed, then the reconcile will be triggered by us

Reply:
WDYT about having both fields and failing the reconciliation? When we implement the validating webhook, we can add that check.

Reply:
I don't see the reason to support both options, it is just confusing, both to us and to the customer. Anyway you will have to support both options in the code, which adds to the confusion. Why are you worried about conversion webhook? I think that at this stage we can just tell customers to move to new version and change their code, i think that any customer we have right now is at very starting stage

Reply:
If we are already breaking the API. Do you have any objection to have [that one](https://github.com/kubernetes-sigs/kernel-module-management/pull/86) as well without a conversion webhook?

Reply:
i am good with that, @qbarrand ?

Reply:
OK to remove the field per @bthurber's decision.
How about changing the new field name to `DockerfileConfigMap`?

Reply:
If we are already breaking the API then let's merge this one as well https://github.com/kubernetes-sigs/kernel-module-management/pull/86.